### PR TITLE
win_get_url: ExtendedWebClient with Timeout support

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
Extended the WebClient object with TimeOut support.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
win_get_url

##### ANSIBLE VERSION
v2.4

### CODE REVIEW

**Comment by jhawkesworth** [comment]
LGTM
really minor nit pick - I would change
$Source something like $extended_webclient_source just to fit with variable naming convention in the rest of the file.  Not a stopper.

**Comment by dagwieers** [comment]
@jhawkesworth Fixed. Thanks !

**Comment by jhawkesworth** [comment]
shipit

File: lib/ansible/modules/windows/win_get_url.ps1 [hunk 1]
```diff
@@ -69,41 +88,40 @@ Function Download-File($result, $url, $dest, $headers, $credentials, $timeout, $
...
     }
 
-    Try {
-        if (-not $whatif) {
-            $webClient.DownloadFile($url, $dest)
+    if (-not $whatif) {
+        Try {
+            $extWebClient.DownloadFile($url, $dest)
+        } Catch [System.Net.WebException] {
+            $result.status_code = $_.Exception.Response.StatusCode
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_get_url.ps1]: You would have to try and cast this as an `[int]`, this returns the human readable enum value and not the number.

Edit: ignore me looks like it is auto converted when json get's to it

**Comment by jborean93** [comment]
Works for me.

shipit

**Pull request merged**